### PR TITLE
[Docs] Add DR to move versions API to MediaCreation

### PR DESCRIPTION
**Pull request by foundrytom**
TL;DR The versioning API (`entityVersion`, `entityVersions`, `getFinalizedEntityVersion`) will be removed from the core API, replaced by formal traits in MediaCreation. Explained with rationale [here](https://github.com/OpenAssetIO/OpenAssetIO/blob/a9d16ed5ce2825c1976e18e06bb8cbd750518b62/doc/decisions/DR017-Entity-version-queries.md#outcome)

Read all about the background and more details [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/962/files?short_path=17bb0c7#diff-17bb0c753b5eaf352dbc0b5d4964c88b0bdf8f0ec14acbf11ab1ae8fe55cbfa6).

### CODE REVIEW

File: doc/decisions/DR017-Entity-version-queries.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  call count.
+- Allows domain specific definitions of a "versions" without needing a
+  core API change.
+- Smaller API surface area to maintain.
+
+#### Cons
+
+- More abstract.
+- Increases API requests when only the version name is required.
+
```
Comment (feltech) [inline doc/decisions/DR017-Entity-version-queries.md:210]: [consider] One more `switch` case in the manager's implementation. But then won't most implementation details be shared between version queries and more generic relationship queries anyway?

**Comment by foundrytom** [comment]
Thanks for documenting our chat @feltech - updated in https://github.com/OpenAssetIO/OpenAssetIO/pull/962/commits/4568a85cf896ff2c10fe1048651128e49604e81c (posting the same link n times on each comment didn't seem helpful)

File: doc/decisions/DR017-Entity-version-queries.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+### Option 1 - Keep first-class versions API
+
+With this option, OpenAssetIO would maintain first-class methods to
+introspect entity versions.
+
+#### Pros
+
+- Optimal in situations where the version name is the only required
+  information to be presented to the suer.
+- Less abstract - defining versions as a first-class API concept.
```
[the remark this answers is not in the document]
Comment (elliotcmorris) [reply doc/decisions/DR017-Entity-version-queries.md]: Question on my mind is why `versions` specifically, seems to my admittedly narrow viewpoint like there's an equal argument that `location` and `displayName` are just as first-class as versions, if we go down that route.

File: doc/decisions/DR017-Entity-version-queries.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+An editorial tool needs to decorate the display of a large number clips
+on a timeline so they can be identified by the user. In this situation,
+separate methods for versioning predicates and additional API call then
+this information is needed.
+
+| Presentation     | Existing                    | Decomposed | Difference |
+| ---------------- | --------------------------- | ---------- | ---------- |
+| Name             | `resolve`                   | `resolve`  | 0          |
+| Name and Version | `resolve` + `entityVersion` | `resolve`  | -1         |
```
[the remark this answers is not in the document]
Comment (foundrytom) [reply doc/decisions/DR017-Entity-version-queries.md:136]: Just that you can add as many traits as you like to one resolve call, but when a separate method is the only way to get version info, it forces you to make two calls.

**Review by feltech — APPROVED** [review]
Just a couple of typos

File: doc/decisions/DR017-Entity-version-queries.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (foundrytom) [reply doc/decisions/DR017-Entity-version-queries.md:141]: Attempted to clarify this [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/962/commits/68e66972f7780b090f18d679613504528b0c8f7f), (typ0 fix [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/962/commits/31e0b525a2a078813f643019a76ee88ac850a4fe)),

File: doc/decisions/DR017-Entity-version-queries.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+  core API should not have strong opinions about domain specific
+  concepts.
+- Allows arbitrary data to be queried along-side versioning information
+  (by expanding the resolved trait set), facilitating runtime
+  customization of the user presentation with constant call count.
+- Smaller API surface area to maintain across language bindings/test
+  infrastructure, and fewer methods for a manager to implement, reducing
+  edge-case handling programming errors.
+- Conveniences can be provided in domain-specific layers that mimic a
+  core API method, discouraging host miss-use.
```
[the remark this answers is not in the document]
Comment (foundrytom) [reply doc/decisions/DR017-Entity-version-queries.md]: (Hopefully) fixed [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/962/commits/4cc2ed59e34f813aa3762e9e415920dc798da468).

File: doc/decisions/DR017-Entity-version-queries.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+With this option, all versioning information would be queried through
+the generic `resolve`, `getWithRelationship` API methods.
+
+#### Pros
+
+- Facilitates domain-specific versioning models through alternate
+  traits, without a core API change. OpenAssetIO axioms dictate that the
+  core API should not have strong opinions about domain specific
+  concepts.
+- Allows arbitrary data to be queried along-side versioning information
```
[the remark this answers is not in the document]
Comment (foundrytom) [reply doc/decisions/DR017-Entity-version-queries.md]: (Hopefully) fixed [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/962/commits/4cc2ed59e34f813aa3762e9e415920dc798da468). Lol, no, and [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/962/commits/37f0b8e526dfd3ed75796a99a719923860d7015a) as my brain clearly isn't working today.

File: doc/decisions/DR017-Entity-version-queries.md [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+| ------------     | --------                   | ----------                      | ---------- |
+| Version          | `entityVersions`           | `getWithRelationship + resolve` | +1         |
+| Version and Date | `entityVersions + resolve` | `getWithRelationship + resolve` | 0          |
+
+#### Annotating a timeline with clip name and version info
+
+An editorial tool needs to decorate the display of a large number clips
+on a timeline so they can be identified by the user. In this situation,
+separate methods for versioning predicates and additional API call then
+this information is needed.
```
[the remark this answers is not in the document]
Comment (foundrytom) [reply doc/decisions/DR017-Entity-version-queries.md]: A tad! You're being very generous.  Hopefully improved [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/962/commits/68e66972f7780b090f18d679613504528b0c8f7f).

**Comment by foundrytom** [comment]
@elliotcmorris @feltech Hopefully reflected all our various conversations and your excellent comments on the PR - thanks. Added a few more pros/cons that weren't already covered. Diff from the original [here](https://github.com/OpenAssetIO/OpenAssetIO/compare/a4a9412..37f0b8e).

File: doc/decisions/DR017-Entity-version-queries.md [hunk 7]
```diff
@@ -0,0 +1,279 @@
+# DR017 Entity version queries
+
+- **Status:** Proposed
+- **Impact:** High
+- **Driver:** @foundrytom @feltech
+- **Approver:** @elliotcmorris @feltech
+- **Contributors:** @foundrytom @feltech @lliotcmorris
```
Comment (feltech) [inline doc/decisions/DR017-Entity-version-queries.md:7]: [fix]
```suggestion
- **Contributors:** @foundrytom @feltech @elliotcmorris
```

File: doc/decisions/DR017-Entity-version-queries.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (elliotcmorris) [reply doc/decisions/DR017-Entity-version-queries.md]: I think you meant [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/962/commits/31e0b525a2a078813f643019a76ee88ac850a4fe)? All good though.

**Comment by elliotcmorris** [comment]
Squash away.
